src/ripple/overlay/impl/PeerImp.cpp

```diff
@@ -360,16 +366,20 @@ PeerImp::json()
 bool
 PeerImp::hasLedger (uint256 const& hash, std::uint32_t seq) const
 {
-    std::lock_guard<std::mutex> sl(recentLock_);
-    if ((seq != 0) && (seq >= minLedger_) && (seq <= maxLedger_) &&
-            (sanity_.load() == Sanity::sane))
-        return true;
-    if (std::find(recentLedgers_.begin(),
-            recentLedgers_.end(), hash) != recentLedgers_.end())
-        return true;
-    return seq >= app_.getNodeStore().earliestSeq() &&
-        boost::icl::contains(shards_,
+    {
+        std::lock_guard<std::mutex> sl(recentLock_);
+        if ((seq != 0) && (seq >= minLedger_) && (seq <= maxLedger_) &&
+                (sanity_.load() == Sanity::sane))
+            return true;
+        if (std::find(recentLedgers_.begin(),
```

Comment:
Maybe just return the value from `std::find` instead of the branch only to return `true`?

Reply:
If not found we want to continue and check the shard store.

Reply:
Ah, good catch. I missed that. Sorry!